### ISSUE TITLE
Unexpected output from get_topic_info()

### ISSUE THREAD

**Issue by drob-xx** [MaartenGr/BERTopic#572]
I'm seeing some small inconsistencies in the output from get_topic_info().

My model params are:

```
{'calculate_probabilities': False,
 'diversity': None,
 'embedding_model': <bertopic.backend._sentencetransformers.SentenceTransformerBackend at 0x7f99cd23e890>,
 'hdbscan_model': HDBSCAN(min_cluster_size=10, prediction_data=True),
 'language': 'english',
 'low_memory': True,
 'min_topic_size': 10,
 'n_gram_range': (1, 1),
 'nr_topics': None,
 'seed_topic_list': None,
 'top_n_words': 20,
 'umap_model': UMAP(angular_rp_forest=True, metric='cosine', min_dist=0.0, n_components=5, tqdm_kwds={'bar_format': '{desc}: {percentage:3.0f}%| {bar} {n_fmt}/{total_fmt} [{elapsed}]', 'desc': 'Epochs completed', 'disable': True}),
 'vectorizer_model': CountVectorizer(ngram_range=(1, 3), stop_words='english'),
 'verbose': True}
```

I'm running on Colab+ with GPU enabled.

There are 354 topics -1 -> 350. When I run:

```
topicDict = BERTNewsModel.get_topic_info().to_dict()['Name']
for idx in range(70,85) :
  print(idx-1, topicDict[idx])
```
I get:

```
69 69_weight_eating_stone_size
70 70_ghana_cup_africa_tournament
71 71_soldier_soldiers_rigby_adebolajo
72 72_tax_osborne_chancellor_mr osborne
73 73_family_mrs_death_died
74 74_blaze_firefighters_crews_smoke
75 78_civil war 2014_war 2014_syrian civil_syrian civil war
76 80_liverpool_rodgers_gerrard_sturridge
77 79_boko_boko haram_haram_nigeria
78 77_shutdown_republicans_tax_senate
79 76_immigration_law_immigrants_arizona
80 75_arsenal_wenger_welbeck_ozil
81 81_lights_northern lights_aurora_northern
82 82_oil_spill_bp_gulf
83 83_isis_syria_foley_islamic
```

Note that my adjusted `idx` value is synced, not synced, and re-synced to the output from `get_topic_info`. This happens over and over again throughout the listings, the numbers go off, then get re-aligned. As far as I can tell this is consistent within the output of get_topic_info() so that:
```

print(BERTNewsModel.get_topic_info().to_dict()['Topic'][76])
print(BERTNewsModel.get_topic_info().to_dict()['Count'][76])
print(BERTNewsModel.get_topic_info().to_dict()['Name'][76])

78
61
78_civil war 2014_war 2014_syrian civil_syrian civil war
```

I encountered this problem when formatting data for a graph and I stepped through all the keys in the returned dict expecting that they would be ordered consistently. I can fix my code to adapt but am still not understanding what I'm seeing.

**Comment by MaartenGr** [comment]
Thank you for sharing this issue! It seems that the indices of the dataframe are mixed up because the number of topics are for several records the same which might have mixed up some of the indices. Fortunately, this is not an issue with the topics of how the model is fitted but merely with representing the dataframe that you get out of `get_topic_info()`. A quick fix is to run the following:

```python
topicDict = BERTNewsModel.get_topic_info().reset_index(drop=True).to_dict()['Name']
for idx in range(70,85) :
    print(idx-1, topicDict[idx])
```

In other words, adding `.reset_index(drop=True)` to the dataframe should solve your issue. I will make sure this is added to BERTopic in the next release, thanks for noticing this!

**Comment by MaartenGr** [comment]
Seeing as this was fixed in v0.11, I will close this issue for now. Having said that, if you still run into this problem, please let me know.
